src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -91,6 +91,7 @@ class LazyVals extends MiniPhaseTransform with IdentityDenotTransformer with Nee
     appendOffsetDefs.get(cls) match {
       case None => template
       case Some(data) =>
+        data.defs.foreach(_.symbol.addAnnotation(defn.ScalaStaticAnnot))
```

Comment:
To avoid the repetition, why not define in LazyVals:

    def staticAnnot = Annotation(defn.ScalaStaticAnnot, Nil)